extensions/wrapper/clients/typescript-client/src/oauth2/OAuthMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+import {Middleware, RequestContext, ResponseContext} from '../generated';
+import {AccessTokenInjector} from './AccessTokenInjector';
+import {AccessTokenStore} from './AccessTokenStore';
+import {needsAuthentication} from './HttpUtils';
+
+export class OAuthMiddleware {
```

Comment:
we might also want to write this code as "real" js code, e.g. only functions

Reply:
that would mean rewriting every class into functions and currently an approach of object instances that fulfill services is used. Is this really necessary?

Reply:
oh no worries, i just thought that this class could be replaced by a simple `export function buildClientCredentialsMiddleware(config: ClientCredentialsConfig): Middleware`, cleaner API